docs/Frequently Asked Questions/React Native/index.md

```diff
@@ -0,0 +1,247 @@
+
+@interface ProxyManager : NSObject <RCTBridgeModule>
+
+<#...#>
+
+@end
+```
+
+###### ProxyManager.m
+An `RCT_EXPORT_MODULE()` macro must be added to the implementation file to expose the class to React Native.
+```objc
+@implementation ProxyManager
+
+RCT_EXPORT_MODULE();
+<#...#>
+
+@end
+```
+
+##### Swift
+First, you must add  `#import "React/RCTBridgeModule.h"` to your `Bridging Header` before you move forward. When creating a Swift application and importing Objective-C code, Xcode should ask if it should create this header file for you. You must include this bridging header for your React Native app to work. You can create this file manually as well. 
+
+```swift
+@objc(ProxyManager)
+
+class ProxyManager: NSObject {
+
+<#...#>
+
+}
+```
+
+Next, to expose the Swift class to React Native you must create an Objective-C file in order to use React Native macros.
+
+##### Objective-C
+###### ProxyManager.m 
+```objc
+#import "React/RCTBridgeModule.h"
+
+@interface RCT_EXTERN_MODULE(ProxyManager, NSObject)
+
+@end
+```
+
+### Post a Notification
+Inside the `ProxyManger` class post a notification for a particular event you wish to execute. The observer of this event will call your React Native lsitener you will set up later in the documentaton below. 
+
+##### Objective-C
+Inside the `ProxyManager` add a soft button to your SDL HMI. Inside the handler post the notification and pass along a reference to the `sdlManager` in order to update the UI. You may choose how to keep a reference to the `sdlManager` object however you like.
+
+```objc
+SDLSoftButtonObject *softButton = [[SDLSoftButtonObject alloc] initWithName:@"Button" state:[[SDLSoftButtonState alloc] initWithStateName:@"State 1" text:@"Data" artwork:nil] handler:^(SDLOnButtonPress * _Nullable buttonPress, SDLOnButtonEvent * _Nullable buttonEvent) {
+    if (buttonPress == nil) { return; }
+
+    NSDictionary *userInfo = @{@"sdlManager": self.sdlManager};
+    [[NSNotificationCenter defaultCenter] postNotificationName:<#Notification Name#> object:nil userInfo:managers];
+}];
+
+self.sdlManager.screenManager.softButtonObjects = @[softButton];
+```
+
+##### Swift
+```swift
+let softButton = SDLSoftButtonObject(name: "Button", state: SDLSoftButtonState(stateName: "State", text: "Data", artwork: nil), handler: { (buttonPress, butonEvent) in
+    guard buttonPress == nil else { return }
+    
+    let userInfo = ["sdlManager": self.sdlManager]
+    NotificationCenter.default.post(name: NSNotification.Name(rawValue: <#Notification Name#>), object: nil, userInfo: managers)
+})
+
+self.sdlManager.screenManager.softButtonObjects = [softButton];
+```
+
+### Create the EventEmitter Class
+
+Create the class that will be the listener for the notiification you created above. This class will be sending an receieving messages from your JavaScript code (React Native). There are some required method(s) you must include inorder to be an  `EventEmitter` , these method(s) are in the example below. 
```

Comment:
```suggestion
Create the class that will be the listener for the notiification you created above. This class will be sending and receiving messages from your JavaScript code (React Native). There are some required method(s) you must include in order to be a React Native `EventEmitter`; these method(s) are in the example below. 
```

Reply:
Is it one method or multiple? If one, then use `method` if multiple, use `methods`. Don't put `method(s)`